### DISCUSSION TITLE
CycloneDDS and eProsima Micro XRCE-DDS Communication in ROS2

### DISCUSSION THREAD

**Discussion by yigitboracagiran** [eProsima/Fast-DDS#5139] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

In default behavior when I send a goal with NAV2(CycloneDDS) and the goal becomes successful state, Nav2 should send a twist message to STM(eProsima Micro XRCE-DDS) which should stop to robot.

### Current behavior

In default behavior when I send a goal with NAV2(CycloneDDS) and the goal becomes successful state, Nav2 should send a twist message to STM(eProsima Micro XRCE-DDS) which should stop to robot. Sometimes Nav2 goal becomes successful sometimes robot turns around itself.

### Steps to reproduce

Send a goal with Nav2, when it becomes successful sometimes robot turns around itself.

### Fast DDS version/commit

eProsima Micro XRCE-DDS: the default micro-ROS middleware

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

I'm currently working on a project on ROS2. I'm using STM32F446ZET6U on my controller circuit and ROS Humble in my computer. MicroROS (STM) is using eProsima Micro XRCE-DDS, Nav2 (ROS2-HUMBLE-NAV2) is using CycloneDDS. 
Note: I don't know for sure if this is happening because of DDS's difference but this is the most reasonable reason.
Note: Another thing I don't know is this could be also happening while robot is also going to somewhere but we can't know for sure because Nav2 sends twist messages all the time. If one message doesn't reach, next message will reach. (There are 10 ms time difference between 2 twist messages.)

**Comment by Mario-DL** [comment]
Hi @yigitboracagiran,

Thanks for using Fast DDS. 
We would need more information on the context, tracebacks or logging info, network setup... 
You could check whether the issue persists using Fast DDS in your ROS 2 Humble environment. 
Since the issue may be related with Micro XRCE, the correct repo to open it would be [Micro XRCE Client](https://github.com/eProsima/Micro-XRCE-DDS-Client) or [Micro XRCE Agent](https://github.com/eProsima/Micro-XRCE-DDS-Agent).
You may also consider contacting [commercials support](https://github.com/eProsima/Fast-DDS/tree/master#getting-help) for premium support options.

For the moment I am moving this to the proper discussion forum.
